File: Java/BinarySearch.java
//WAP in JAVA to find a value from array using binary search
public class BinarySearch 
{
    public static void main(String[] args) 
    {
        int[] arr = {1,2,3,4,5,6,7,8,9,10};
        int target = 8;
        int ans = binarySearch(arr, target);
        int ans2 = ans+1;
        System.out.println("The element is found at= "+ ans2 +" number position");
    }
    // return the index
    // return -1 if it does not exist
    static int binarySearch(int[] arr, int target) 
    {
        int start = 0;
        int end = arr.length - 1;
        while(start <= end) 
        {
            // find the middle element
            // might be possible that (start + end) exceeds the range of int in java
            int mid = start + (end - start) / 2;
            if (target < arr[mid]) 
            {
                end = mid - 1;
            }
            else if (target > arr[mid]) 
            {
                start = mid + 1;
            } 
            else 
            {
                //when answer found at middle
                return mid;
            }
        }
        return -1;
    }
}

File: javascript/binarySearch.js
// Iterative Binary Serach Function implemented in JavaScript
/**
 * 
 * @param {Array to be searched in} arr 
 * @param {elemnt to be searched} target 
 * @returns index of target if found, else -1;
 */

function binarySearchIterative(arr, target){
    let left = 0;
    let right = arr.length - 1;
    let mid;
    while ( left <= right ){
        mid = Math.floor( ( right + left ) /2 );
        if ( arr[mid] == target ){
            return mid;
        } else if ( arr[mid] < target ){
            left = mid + 1;
        }
        else {
            right = mid - 1;
        }
    }
    return -1;
}


// Test
const arr = [12, 24, 32, 39, 45, 50, 54]  
const target = 12 
    
const result = binarySearchIterative(arr, target)  
  
if ( result != -1){
    console.log("Element is present at index", result)  
}  
else{
    console.log("Element is not present in arr") 
}